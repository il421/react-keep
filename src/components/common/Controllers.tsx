import React from "react";
import { AlignItems, JustifyContent } from "../../common";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/common/_controllers.scss";
import { NoteType } from "../notes";

interface Controllers extends Partial<Pick<HTMLElement, "className">> {
  isMobile?: boolean;
  openDialog: (type: NoteType) => void;
}

export const Controllers: React.FunctionComponent<Controllers> = ({
  className,
  isMobile = false,
  openDialog,
}) => {
  const handleOnClick = (type: NoteType) => () => openDialog(type);

  return (
    <div
      className={`controllers  ${
        isMobile && "controllers--mobile"
      } ${className}`}
    >
      <FlexBox
        vertical={isMobile}
        justifyContent={JustifyContent.spaceBetween}
        alignItems={AlignItems.center}
      >
        <IconButton
          className="controllers__text"
          onButtonClick={handleOnClick(NoteType.text)}
          icon="align-left"
          size="2x"
        />

        <IconButton
          className="controllers__list"
          onButtonClick={handleOnClick(NoteType.list)}
          icon="check-square"
          size="2x"
        />

        <IconButton
          className="controllers__image"
          onButtonClick={handleOnClick(NoteType.image)}
          icon="image"
          size="2x"
        />
      </FlexBox>
    </div>
  );
};
