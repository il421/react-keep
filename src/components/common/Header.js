import React from 'react';
import { connect } from 'react-redux';

import { startLogout } from '../../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Search from './Search';

export const Header = ({ startLogout, auth, showSidebar }) => {

  return (
    <header className="header" onClick={(evt) => evt.stopPropagation()}>
      <div className="content-container">
        <div className="header__content">
          <div className="header__sidebar">

            <button className="button button--sidenav" onClick={() => showSidebar()}>
              <FontAwesomeIcon icon="bars" size="2x" />
            </button>

          </div>
          <div>
            <h1 className="header__title">Keep Me</h1>
          </div>
          <div className="header__search">
            <Search />
          </div>
          <div className="header__user-box user-box">
            <div className="user-box__photo">
              <img src={ auth.url } width={ 50 } height={ 50 } />
            </div>
            <div className="user-box__name show-for-mobile">{ auth.name }</div>
            <button className="user-box__logout button button--sidenav" onClick={ startLogout }>
              <FontAwesomeIcon icon="sign-out-alt" size="lg" />
            </button>
          </div>
        </div>
      </div>
    </header>
  );
};


const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
    url: state.url,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
