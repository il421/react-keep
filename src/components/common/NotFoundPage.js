import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div className="not-found">
    <div className="not-found__wrapper">
      <div className="not-found__text">
        Sorry, such a page does not exist in the App. <br /><br />
        Try to find something interesting on the Dashboard page.
      </div>
      <Link to="/" className="not-found__link button">Keep me ...</Link>
    </div>
  </div>
);

export default NotFoundPage;
