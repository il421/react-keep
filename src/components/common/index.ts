export * from "./Controllers";
export * from "./DashboardPage";
export * from "./Header";
export * from "./NotFoundPage";
export * from "./Search";
export * from "./SideBar";
