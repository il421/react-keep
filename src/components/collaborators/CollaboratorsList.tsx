import React from "react";
import { FlexBox } from "../ui-components";
import { JustifyContent } from "../../common";
import { CollaboratorsStoreState, Store } from "../../store/store.types";
import { ThunkDispatch } from "redux-thunk";
import { connect } from "react-redux";
import { CollaboratorsItem } from "./CollaboratorsItem";
import { handleRemoveCollaborator } from "../../actions/collaborators";
import "../../styles/components/collaborators/_collaborators-list.scss";

interface TagsListProps {}

interface StateProps {
  collaborators: CollaboratorsStoreState[];
}

interface DispatchProps {
  removeCollaborator: (id: string) => void;
}

export type Props = DispatchProps & StateProps;

export class CollaboratorsListBase extends React.PureComponent<Props> {
  render() {
    return (
      <FlexBox
        vertical
        justifyContent={JustifyContent.start}
        className="collaborators-list"
      >
        {this.props.collaborators.map((coll) => (
          <CollaboratorsItem
            key={coll.uid}
            collaborator={coll}
            removeCollaborator={(id: string) => {
              this.props.removeCollaborator(id);
            }}
          />
        ))}
      </FlexBox>
    );
  }
}

const mapStateToProps = (state: Store): StateProps => {
  return {
    collaborators: state.collaborators,
  };
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  removeCollaborator: (id: string) => dispatch(handleRemoveCollaborator(id)),
});

export const CollaboratorsList = connect<
  StateProps,
  DispatchProps,
  TagsListProps,
  Store
>(
  mapStateToProps,
  mapDispatchToProps
)(CollaboratorsListBase);
