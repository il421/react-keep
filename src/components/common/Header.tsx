import React from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import { stringify } from "query-string";
import { AuthStoreState, Store } from "../../store/store.types";
import { QueryKeys, RouteActions } from "../../routers/Routing";
import { startLogout } from "../../actions/auth";
import Search from "./Search";
import { Controllers } from "./Controllers";
import {
  IconButton,
  UserPhoto,
  ContentContainer,
  LinkButton
} from "../ui-components";
import "../../styles/components/header/_header.scss";
import "../../styles/components/header/_user-box.scss";
import "../../styles/components/common/_visibility.scss";
import { ThunkDispatch } from "redux-thunk";

interface HeaderProps {
  showSidebar: (value: boolean) => void;
}

interface StateProps {
  auth: AuthStoreState;
}

interface DispatchProps {
  startLogout: () => void;
}

type Props = HeaderProps & DispatchProps & StateProps;

const Header: React.FunctionComponent<Props> = ({
  startLogout,
  auth,
  showSidebar
}) => {
  const history = useHistory();

  const onClickHandler = (key: keyof typeof QueryKeys) => {
    const query = stringify({
      [key]: RouteActions.edit
    });
    history.push(`${history.location.pathname}?${query}`);
  };

  return (
    <header className="header" onClick={evt => evt.stopPropagation()}>
      <ContentContainer>
        <div className="header__content">
          <IconButton
            className="header__bar"
            icon="bars"
            size="2x"
            onClick={() => showSidebar(true)}
          />

          <h1 className="header__title">Keep Me</h1>

          <Search wrapperClass="header__search" />

          <Controllers className="hide-for-mobile" />

          <div className="header__user-box user-box">
            {auth.name && auth.url && (
              <UserPhoto
                className="user-box__photo"
                src={auth.url}
                height={50}
                width={50}
              />
            )}
            <LinkButton
              text={auth.name ?? "Update profile"}
              type="button"
              className="user-box__link"
              onClick={() => onClickHandler(QueryKeys.user)}
            />
            <IconButton
              className="user-box__logout"
              icon="sign-out-alt"
              size="lg"
              onClick={startLogout}
            />
          </div>
        </div>
      </ContentContainer>
    </header>
  );
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state: Store): StateProps => {
  return {
    auth: state.auth
  };
};

export default connect<StateProps, DispatchProps, HeaderProps, Store>(
  mapStateToProps,
  mapDispatchToProps
)(Header);
