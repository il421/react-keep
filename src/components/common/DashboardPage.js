import React, { useState } from 'react';

import NoteSelection from '../note/NoteSelection';
import NotesList from '../note/NotesList';
import Header from './Header';
import SideBar from './SideBar';


const DashboardPage = () => {
  let [showBar, toggleSidebarDisplay] = useState(false);

  const showSidebar = () => {
    toggleSidebarDisplay(showBar = true);
  };

  const hideSidebar = () => {
    toggleSidebarDisplay(showBar = false);
  };

  return (
    <div className="dashboard">
      <Header showSidebar={ showSidebar } />
      <NoteSelection className="content-container"/>
      <NotesList/>

      <SideBar showBar={ showBar } hideSidebar={ hideSidebar } />
    </div>
  );
};
export default DashboardPage;
