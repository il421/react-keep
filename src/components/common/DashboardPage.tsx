import React from "react";
import { ToastContainer } from "react-toastify";
import { History, Path } from "history";
import { JustifyContent } from "../../common";
import { Header } from "./Header";
import { SideBar } from "./SideBar";
import { Controllers } from "./Controllers";
import UserFormModal from "../login/UserFormModal";
import "../../styles/components/common/_dashboard.scss";
import "../../styles/components/notes/_note-modal.scss";
import { QueryKeys } from "../../routers/Routing";
import { stringify } from "query-string";
import {
  NoteType,
  ImageNoteFormModal,
  TextNoteFormModal,
  ListNoteFormModal,
} from "../notes";
import { FlexBox } from "../ui-components";
import { NotesList } from "../notes/NotesList";

export interface DashboardPageProps {
  history: History;
}

export const onNoteSelected = (options: {
  type: NoteType;
  id: string;
  pathname: string;
  push: (path: Path) => void;
}) => {
  let key: keyof typeof QueryKeys;
  const { type, id, pathname, push } = options;
  switch (type) {
    case NoteType.text:
      key = QueryKeys.text;
      break;
    case NoteType.list:
      key = QueryKeys.list;
      break;
    default:
      key = QueryKeys.image;
  }

  const query = stringify({
    [key]: id,
  });
  push(`${pathname}?${query}`);
};

export const DashboardPage: React.FunctionComponent<DashboardPageProps> = ({
  history,
}): JSX.Element => {
  return (
    <FlexBox
      vertical={true}
      justifyContent={JustifyContent.spaceBetween}
      className="dashboard"
    >
      <Header history={history} />
      <NotesList
        onNoteSelected={(type: NoteType, id: string) =>
          onNoteSelected({
            type,
            id,
            pathname: history.location.pathname,
            push: history.push,
          })
        }
      />

      <SideBar />
      <TextNoteFormModal history={history} />
      <ListNoteFormModal history={history} />
      <ImageNoteFormModal history={history} />
      <UserFormModal />
      <Controllers
        isMobile
        openDialog={(query) =>
          history.push(`${history.location.pathname}?${query}`)
        }
      />

      <ToastContainer autoClose={2000} />
    </FlexBox>
  );
};
