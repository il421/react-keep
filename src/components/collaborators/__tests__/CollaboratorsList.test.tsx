import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { Props, CollaboratorsListBase } from "../CollaboratorsList";

import { collaborators } from "../../../testData/users";
import { WithFontAwesome } from "../../../common/WithFontAwesome";

let props: Props, wrapper: ReactWrapper<Props>;

beforeEach(() => {
  props = {
    collaborators,
    removeCollaborator: jest.fn(),
  };
  wrapper = mount<Props>(
    <WithFontAwesome>
      <CollaboratorsListBase {...props} />
    </WithFontAwesome>
  );
});

test("should render CollaboratorsList", () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

test("should call removeCollaborator if click the remove button", () => {
  wrapper.find("button").first().simulate("click");
  expect(props.removeCollaborator).toHaveBeenLastCalledWith(
    collaborators[0].uid
  );
});
