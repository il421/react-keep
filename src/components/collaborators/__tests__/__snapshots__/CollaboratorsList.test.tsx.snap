// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CollaboratorsList 1`] = `
"<WithFontAwesome>
  <CollaboratorsListBase collaborators={{...}} removeCollaborator={[Function: mockConstructor]}>
    <FlexBox vertical={true} justifyContent=\\"flex-start\\" className=\\"collaborators-list\\">
      <div style={{...}} className=\\"collaborators-list\\">
        <CollaboratorsItem collaborator={{...}} removeCollaborator={[Function: removeCollaborator]}>
          <FlexBox justifyContent=\\"space-between\\" alignItems=\\"center\\" className=\\"collaborators-item\\">
            <div style={{...}} className=\\"collaborators-item\\">
              <FlexBox justifyContent=\\"flex-start\\">
                <div style={{...}} className={[undefined]}>
                  <Memo(CoinBase) email=\\"test1@gmail.com\\" name=\\"Ivan\\" url=\\"http://test.com\\">
                    <div data-tip=\\"Ivan\\">
                      <FlexBox justifyContent=\\"center\\" className=\\"coin undefined\\">
                        <div style={{...}} className=\\"coin undefined\\">
                          <img src=\\"http://test.com\\" alt=\\"Photo of Ivan\\" width={25} height={25} />
                        </div>
                      </FlexBox>
                    </div>
                  </Memo(CoinBase)>
                  <div className=\\"collaborators-item__value\\">
                    Ivan
                  </div>
                </div>
              </FlexBox>
              <IconButton icon=\\"times\\" onButtonClick={[Function]}>
                <button id={[undefined]} className=\\"icon-button undefined \\" onClick={[Function]} disabled={[undefined]} type=\\"button\\">
                  <FontAwesomeIcon icon=\\"times\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                    <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"times\\" className=\\"svg-inline--fa fa-times fa-w-11 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 352 512\\" style={{...}}>
                      <path fill=\\"currentColor\\" d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" style={{...}} />
                    </svg>
                  </FontAwesomeIcon>
                </button>
              </IconButton>
            </div>
          </FlexBox>
        </CollaboratorsItem>
        <CollaboratorsItem collaborator={{...}} removeCollaborator={[Function: removeCollaborator]}>
          <FlexBox justifyContent=\\"space-between\\" alignItems=\\"center\\" className=\\"collaborators-item\\">
            <div style={{...}} className=\\"collaborators-item\\">
              <FlexBox justifyContent=\\"flex-start\\">
                <div style={{...}} className={[undefined]}>
                  <Memo(CoinBase) email=\\"test2@gmail.com\\" name=\\"Tany\\" url=\\"http://tdest.com\\">
                    <div data-tip=\\"Tany\\">
                      <FlexBox justifyContent=\\"center\\" className=\\"coin undefined\\">
                        <div style={{...}} className=\\"coin undefined\\">
                          <img src=\\"http://tdest.com\\" alt=\\"Photo of Tany\\" width={25} height={25} />
                        </div>
                      </FlexBox>
                    </div>
                  </Memo(CoinBase)>
                  <div className=\\"collaborators-item__value\\">
                    Tany
                  </div>
                </div>
              </FlexBox>
              <IconButton icon=\\"times\\" onButtonClick={[Function]}>
                <button id={[undefined]} className=\\"icon-button undefined \\" onClick={[Function]} disabled={[undefined]} type=\\"button\\">
                  <FontAwesomeIcon icon=\\"times\\" border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} swapOpacity={false}>
                    <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"times\\" className=\\"svg-inline--fa fa-times fa-w-11 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 352 512\\" style={{...}}>
                      <path fill=\\"currentColor\\" d=\\"M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z\\" style={{...}} />
                    </svg>
                  </FontAwesomeIcon>
                </button>
              </IconButton>
            </div>
          </FlexBox>
        </CollaboratorsItem>
      </div>
    </FlexBox>
  </CollaboratorsListBase>
</WithFontAwesome>"
`;
